src/nvim/eval.c

```diff
@@ -15577,6 +15577,53 @@ static void f_split(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   p_cpo = save_cpo;
 }
 
+/// "stdpath()" helper for list results
+void stdpath_list_helper(const XDGVarType xdg, typval_T *rettv)
+{
+  const void *iter = NULL;
+  list_T *const list = tv_list_alloc();
+  rettv->v_type = VAR_LIST;
+  rettv->vval.v_list = list;
+  list->lv_refcount++;
+  char *const dirs = stdpaths_get_xdg_var(xdg);
+  do {
+    size_t dir_len;
+    const char *dir;
+    iter = vim_colon_env_iter(dirs, iter, &dir, &dir_len);
+    if (dir != NULL && dir_len > 0) {
+      char *dir_with_nvim = xmemdupz(dir, dir_len);
+      dir_with_nvim = concat_fnames_realloc(dir_with_nvim, "nvim", true);
+      tv_list_append_string(list, dir_with_nvim, strlen(dir_with_nvim));
+      xfree(dir_with_nvim);
+    }
+  } while (iter != NULL);
+  xfree(dirs);
+}
+
+/// "stdpath(type)" function
+static void f_stdpath(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
```

Comment:
I would actually suggest to assign `v_type = VAR_STRING` and `vval.v_string = NULL` here to get consistent results on error returns “type error” and “unknown path requested”.